test/api/group_sets_api_test.rb

```diff
@@ -0,0 +1,44 @@
+require 'test_helper'
+require 'user'
+
+class GroupSetsApiTest < ActiveSupport::TestCase
+  include Rack::Test::Methods
+  include TestHelpers::AuthHelper
+  include TestHelpers::JsonHelper
+  include TestHelpers::TestFileHelper
+
+  def app
+    Rails.application
+  end
+
+  def test_get_all_groups_in_unit_with_authorization
+    # Create a group
+    newGroup = FactoryBot.create(:group)
+    # Obtain the unit of the group
+    newUnit = newGroup.group_set.unit
+    get with_auth_token "/api/units/#{newUnit.id}/groups",newUnit.main_convenor_user
+    
+    #Check response
+    response_keys = %w(id name)
+    last_response_body.each do | data |
+      grp = Group.find(data['id'])
+      assert_json_matches_model(data, grp, response_keys)
+      assert_equal data['name'],grp['name']
```

Comment:
You already check for `name` attribute line 25 assertion, why bother checking again?